insights/src/main/kotlin/com/algolia/instantsearch/insights/event/EventWorker.kt

```diff
@@ -0,0 +1,35 @@
+package com.algolia.instantsearch.insights.event
+
+import androidx.work.Data
+import androidx.work.Worker
+import com.algolia.instantsearch.insights.Insights
+import com.algolia.instantsearch.insights.webservice.uploadEvents
+
+
+internal class EventWorker : Worker() {
+
+    private enum class Keys {
+        Index
+    }
+
+    companion object {
+
+        fun buildInputData(indexName: String): Data {
+            return Data.Builder().putAll(
+                mapOf(Keys.Index.name to indexName)
+            ).build()
+        }
+
+        fun Data.getInputData(): String {
+            return getString(Keys.Index.name, null)
+        }
+    }
+
+    override fun doWork(): WorkerResult {
+        val indexName = inputData.getInputData()
+        val insights = Insights.shared(indexName)
+        val failedEvents = insights.webService.uploadEvents(insights.database, indexName)
+
+        return if (failedEvents.isEmpty()) WorkerResult.SUCCESS else WorkerResult.RETRY
```

Comment:
The ternary operator no longer exists for such case is Kotlin, because now you can use if as an expression rather than just a statement.
This means you can assign the value returned by an if statement, provided all possible cases are covered (with an else, usually).

In this case, the function doWork is returning a `WorkerResult` value from an if / else expression, based on the `failedEvents.isEmpty()` condition check.

The `?:` operator still exists, but only for checking nullability of a variable, providing a default fallback, and unboxing the value into a non-null type.

Example:

```
val nullable: String? = null

val string: String = nullable ?: "default"
```